### ISSUE TITLE
Oscar 5.0 nightly on fc5

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#271]
**Reported by njpatel on 18 Oct 2006 02:08 UTC**
Hi
All,

I have been trying to install nightly oscar5.0 and here are the file versions I am using

1)  oscar-base-5.1a1r5387-20060930.tar.gz
2)  oscar-repo-common-rpms-5.1a1r5387-20060930.tar.gz
3)  oscar-repo-fc-5-x86_64-5.1a1r5387-20060930.tar.gz

I constanlty get stuck on could not bootstrap smart package manager!!!

Look bellow for the logs...

Can any one help me figure out the issue.

Many Thanks
Nishit Patel

=============================================================================
== Running OSCAR install_cluster script    Tue Oct 17 21:52:58 EDT 2006

=============================================================================

--> Checking if this is a supported platform
--> Running: "cd /opt/oscar/scripts && ./wizard_prep"

=============================================================================
== Running OSCAR wizard_prep script
=============================================================================

Distro repository /tftpboot/distro/fedora-5-x86_64 not found. Creating empty dir ectory.
ERROR: The distribution repository /tftpboot/distro/fedora-5-x86_64 looks empty!
!!!! Copy all your distro package files (rpms) to it and retry.
Oscar Wizard preparation script failed to complete at ./install_cluster line 228 .
[oscar](root@dice)# ./install_cluster eth1
--> Setting env variable OSCAR_UI to gui
Checking your system...
Success: network-check.pl
Success: su-check.pl
Success: debrepo-check.pl
Success: ssh-check.pl
Success: selinux-check.pl
System checked

=============================================================================
== Running OSCAR install_cluster script    Tue Oct 17 21:59:03 EDT 2006

=============================================================================

--> Checking if this is a supported platform
--> Running: "cd /opt/oscar/scripts && ./wizard_prep"

=============================================================================
== Running OSCAR wizard_prep script
=============================================================================

--> Bootstrapping OSCAR smart package manager
Preparing...                ##################################################
        package createrepo-0.4.4-0.2 (which is newer than createrepo-0.4.3-5.1e)  is already installed
install failed: Bad file descriptor
 at /opt/oscar/scripts/install_prereq line 249
        main::handle_dumb_rpms(0, 'perl-IO-Tty', 'python-urlgrabber', 'python-el ementtree', 'yum', 'createrepo-0.4.3-5.1e', 'yume', 'packman-depman', 'packman-d epman-rpms', ...) called at /opt/oscar/scripts/install_prereq line 159
Could not bootstrap smart package manager!
Oscar Wizard preparation script failed to complete at ./install_cluster line 228

**Comment by dikim33** [comment]
**Comment by Steven Blackburn on 28 Oct 2006 15:12 UTC**
From the names of the files, it looks like these are from trunk rather than the branch. You might have better luck from the branch, which will become Oscar 5.0 (the trunk will become 5.1?).

The bug sounds similar to a problem I had trying to install Oscar on an updated (or respun) Fedora Core 5. See ticket #209 which covers this. Either use a raw FC5 or update the prereq.cfg file.

Does that make this ticket a duplicate?

**Comment by dikim33** [comment]
**Comment by efocht on 30 Oct 2006 19:01 UTC**
Hi,

please edit packages/yume/prereq.cfg and delete the version information from the lines containing createrepo. Or replace the file with the prereq.cfg attached to this ticket.

Installing with an updated system requires that your packages repository is up to date (same RPMs as installed on the system). One simple way to achieve this is to create a file /tftpboot/distro/fedora-5-x86_64.url (or ...i386...) with content like below:

```
file:/tftpboot/distro/fedora-5-x86_64
http://ftp-stud.fht-esslingen.de/pub/Mirrors/fedora.redhat.com/linux/core/updates/5/x86_64/
```

The first line is your local (non-updated) RPMs repository. The second line points to a mirror site with updates to fedora core 5.

Regards,
Erich

**Comment by dikim33** [comment]
**Comment by njpatel@fastmail.fm on 31 Oct 2006 02:06 UTC**
Thank you.  It worked for me

**Comment by dikim33** [comment]
**Comment by efocht on 31 Oct 2006 12:57 UTC**
Checked in to trunk (r5486) and branch (r5487).

**Comment by dikim33** [comment]
**Comment by efocht on 31 Oct 2006 12:57 UTC**
(In [5487]) Merged commit r5486 by efocht from 2006-10-31 13:54:12 +0100 (Tue, 31 Oct 2006)
with svnmerge_1by1 version v1.1 31.7.2006
from repository https://svn.oscar.openclustergroup.org/svn/oscar/trunk
to repository https://svn.oscar.openclustergroup.org/svn/oscar/branches/branch-5-0
Original commit message was:

Createrepo doesn't need version info any more. As long as we have
no version comparisons and checking in install_prereq, we should
get rid of this. 
The fixed version string leads to trouble when installing OSCAR
on updated systems which have a newer createrepo rpm installed.
Fix for bug #271.
Closes #271
